src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/IConnectionPlugin.java

```diff
@@ -28,21 +28,32 @@
 
 import com.mysql.cj.conf.ConnectionUrl;
 
+import java.lang.reflect.Method;
 import java.sql.SQLException;
+import java.util.List;
 import java.util.concurrent.Callable;
 
 /**
  * Interface for connection plugins. This class implements ways to execute a JDBC method
  * and to clean up resources used before closing the plugin.
  */
 public interface IConnectionPlugin extends ITransactionContextHandler {
-  Object execute(
+  /**
+   * Execute a JDBC method on a Connection-bound Object such as a Statement or ResultSet.
+   */
+  Object executeOnConnectionBoundObject(
```

Comment:
What's the issue you're trying to solve by making such major (breaking change) change in a public interface?

Reply:
My mistake, I thought that the original execute function did not allow for the connection to be switched out by the plugins, but after testing I think I was wrong. I'll fix this today